parsons/box/box.py

```diff
@@ -0,0 +1,200 @@
+"""Class for Box API.
+
+To get authentication info (this eventually belongs in the docs for
+this connector):
+
+Information on Box site here:
+https://developer.box.com/guides/applications/custom-apps/oauth2-setup/
+
+1. Set up Box account
+2. Go to Developer console (https://app.box.com/developers/console)
+3. Either select an existing app or create a new one. To create a new one:
+  - Select "New App" > "Custom App" > "OAuth 2.0 with JWT"
+4. View your app and select "Configuration" in the left menu
+5. Scroll down to get the client id & secret, and just above it
+   select OAuth2.0 with JWT (Server Authentication) and
+   generate a developer token, aka "access token".
+
+"""
+
+import logging
+
+import boxsdk
+
+from parsons.etl.table import Table
+from parsons.utilities.check_env import check as check_env
+from parsons.utilities.files import create_temp_file
+
+import tempfile
+
+logger = logging.getLogger(__name__)
+
+
+class Box(object):
```

Comment:
Notice that some of the functions have the default id of "0" in the signature, some don't. Can we make it consistent across all functions.

Reply:
I think I've put in defaults (specifically, using the default folder) wherever possible. This makes sense for some methods, e.g. for list_files() and upload_table() - if they've not otherwise specified list files in the default folder, or upload to the default folder, unless otherwise specified.

With other things like delete_file(), it doesn't make sense to have a default file_id.

I've created a global `DEFAULT_FOLDER_ID = '0'` and swapped it in on the function signatures to clarify the semantics.

Reply:
Was looking at line 77, which sets the default at the signature to `None` then do a parent_folder_id = parent_folder_id or '0' inside the body.

Reply:
Ah - yup! I'm sure I had a reason for doing that, but have no idea what it was, and replaced it with parent_folder_id=DEFAULT_FOLDER_ID in the signature.